src/averageAges.js

```diff
@@ -37,7 +49,13 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothersList = people.map(human => human.mother);
+
+  const filteredWoman = !withChildren
+    ? people.filter(human => human.sex === 'f')
+    : people.filter(human => mothersList.includes(human.name));
```

Comment:
So in worst case this array will be iterated 3 times, but this amount can be reduced to 2
No need to map people array, just use `find` or `some` instead of `includes` to get access to object

Also use only one filter and choose needed condition using ternary operator inside it